### PR TITLE
feat: stabilize relayer flow

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: relayer/create_ibc_channel.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"github.com/dymensionxyz/roller/cmd/consts"
+	"github.com/dymensionxyz/roller/cmd/utils"
+)
+
+// Creates an IBC channel between the hub and the client, and return the source channel ID.
+func (r *Relayer) CreateIBCChannel(logFileOption utils.CommandOption) (ConnectionChannels, error) {
+	//TODO: add support for --override flag
+	createClientsCmd := r.getCreateClientsCmd()
+	fmt.Println("💈 Creating clients...")
+	if err := utils.ExecBashCmd(createClientsCmd, logFileOption); err != nil {
```
Comment (ItayLevyOfficial) [inline relayer/create_ibc_channel.go:19]: Using exechbashcmd here can cause issues due to its propensity to crash, and the vague 'exit status 1' error message provides little insight. Consider modifying exechbashcmd to return stderr upon error or alternatively, switch to using ExecBashCmdWithOSOutput, which already implements this feature

File: cmd/utils/bash_commands.go [hunk 2]
```diff
@@ -89,7 +96,7 @@ func ExecBashCommandWithStdErr(cmd *exec.Cmd) (bytes.Buffer, error) {
 	cmd.Stdout = &stdout
 	err := cmd.Run()
 	if err != nil {
-		return stdout, fmt.Errorf("command execution failed: %w, stderr: %s", err, stderr.String())
+		return stderr, fmt.Errorf("command execution failed: %w, stderr: %s", err, stderr.String())
```
Comment (ItayLevyOfficial) [inline cmd/utils/bash_commands.go:99]: why? the stderr is within the error message

File: relayer/channels.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	Counterparty   Counterparty `json:"counterparty"`
+	ConnectionHops []string     `json:"connection_hops"`
+	Version        string       `json:"version"`
+}
+
+type ProofHeight struct {
+	RevNumber string `json:"revision_number"`
+	RevHeight string `json:"revision_height"`
+}
+type HubQueryResult struct {
```
Comment (ItayLevyOfficial) [inline relayer/channels.go:89]: Please retain only the fields that are in actual use

**Review by ItayLevyOfficial — COMMENTED** [review]
Great refactoring work! I've left a few comments

File: relayer/channels.go (hunk 3 above)
Comment (mtsitrin) [reply relayer/channels.go:89]: I think it's more clear this way

File: relayer/create_ibc_channel.go (hunk 1 above)
Comment (mtsitrin) [reply relayer/create_ibc_channel.go:19]: I don't want it to print to stdout, just to the logger

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM

**Pull request merged**